boards/arm/atsamd21_xpro/atsamd21_xpro.dts

```diff
@@ -26,6 +26,14 @@
 		i2c-0 = &sercom2;
 	};
 
+	feature {
+		gpio-loopback {
+			compatible = "zephyr,feature,gpio-loopback";
+			out-gpios = <&portb 6 0>; /* EXT1.5 */
+			in-gpios = <&portb 7 0>;  /* EXT1.6 */
+		};
```

Comment:
I wonder if this is really the way to go.

Instead of modifying board.dts file that is supposed to be application agnostic, 
can we instead provide a test-features.overlay (in same board folder) that would hols these samples specifics settings and that would be enabled only when needed by applications (tests in current case) ?

Reply:
I agreed with @erwango . I was thinking about why not define a "shield" for tests, with proper overlays.

This moves things from one place to another but not help to decrease the redundancies. The majority of the boards have some standard header. With a shield at least tests can benefit from everything that is already been made to decrease maintenance.

board.yaml already said what board support.

supported:
  - arduino_gpio
  - arduino_i2c

At least for those boards, with standard headers, somehow test could run without changes or files.

Reply:
> can we instead provide a test-features.overlay

I'm not sure how that would work.  How should the test ask for the overlay, and will the build infrastructure behave acceptably if one doesn't exist?  One approach is to put:

```
#ifdef DTS_INCLUDE_FIXTURES
#include <fixtures.dtsi>
#endif /* DTS_INCLUDE_FIXTURES */
```

in the board file and use `DTS_EXTRA_CPPFLAGS=-DDTS_INCLUDE_FIXTURES` in the test, but that still requires modifying the board DTS file.

> This moves things from one place to another but not help to decrease the redundancies.

I think it does: now things that need a GPIO loopback can just use one pre-defined for the board, rather than having to copy overlays from one test-specific directory to another.  I agree it's not ideal, though.

> At least for those boards, with standard headers, somehow test could run without changes or files.

It can't, really, because different boards need to use different Arduino pins for the role.  Also not all boards do or should define Arduinio maps (especially if they don't actually have a real Arduino header.  IMO).  For the GPIO loopback case suitable pins need to be identified; for simplicity they are defined in terms of Arduino pins where those are available, but the tests can't find them without a binding that identifies their role.

Reply:
>One approach is to put [] but that still requires modifying the board DTS file.

We can bury that in `skeleton.dtsi`, but I'm not sure if this is acceptable.
Can't we use some cmake magic, that would be based on a flag set in test, to detect that board folder holds such an overlay  ?

Reply:
I have no idea how to do that, but it's probably possible.

Reply:
> can we instead provide a test-features.overlay (in same board folder) that would hold these samples specific settings and that would be enabled only when needed by applications (tests in current case) ? 
can we use some cmake magic, that would be based on a flag set in test, to detect that board folder holds such an overlay ?

@tejlmand does it sounds doable?

Reply:
It is clear that something will be repeated in some place. It can be repeated on the tests by overlays or in boards dts etc. I would explore something to avoid this situation. Not sure it is viable.

I was thinking on the possibility to have a internal directory inside tests to shares the common parts. The below fragment I saw it is repeated for every board with an Arduino Shield header:

```
	feature {
		gpio-loopback {
			compatible = "zephyr,feature,gpio-loopback";
			out-gpios = <&arduino_header 6 0>; /* Arduino D0 */
			in-gpios = <&arduino_header 7 0>;  /* Arduino D1 */
		};
	};
``` 

If with some glue we can reuse this for those boards with a standard Arduino header, for instance, we can avoid the copy paste. This could be repeated for others standard headers too. If with some luck the tests stay inside the Arduino header, for instance, we can avoid for those boards have the fixtures.dtsi file. I think this is what the shields try to do with overlays.

It is clear that any particular cases will be defined at board level but maybe common things could have a common point.

Reply:
The below fragment I saw it is repeated for every board with an Arduino Shield header:

No, it's not.  Many have D0 and D1,  A significant number of A0 and A1.  Some have A4 and A5.  Not to say some sharing can't be accomplished, but it may not be worth the complexity.

Reply:
Yah! I looked the initials and assumed that it were for every board, like serial port D0/D1. Now it is clear the why.
Sorry about the noise.